sdcm/stress/base.py

```diff
@@ -0,0 +1,113 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2019 ScyllaDB
+
+import logging
+import random
+import concurrent.futures
+
+from sdcm.cluster import BaseLoaderSet
+from sdcm.utils.common import generate_random_string
+
+LOGGER = logging.getLogger(__name__)
+
+
+class DockerBasedStressThread:
+    # pylint: disable=too-many-instance-attributes
+    def __init__(self, loader_set, stress_cmd, timeout, stress_num=1, node_list=None,  # pylint: disable=too-many-arguments
+                 round_robin=False, params=None, stop_test_on_failure=True):
+        self.loader_set: BaseLoaderSet = loader_set
+        self.stress_cmd = stress_cmd
+        self.timeout = timeout
+        self.stress_num = stress_num
+        self.node_list = node_list or []
+        self.round_robin = round_robin
+        self.params = params or {}
+        self.loaders = []
+
+        self.executor = None
+        self.results_futures = []
+        self.max_workers = 0
+        self.shell_marker = generate_random_string(20)
+        self.shutdown_timeout = 180  # extra 3 minutes
+        self.stop_test_on_failure = stop_test_on_failure
+
+    def configure_executer(self):
+
+        if self.round_robin:
+            self.stress_num = 1
+            loaders = [self.loader_set.get_loader()]
+            LOGGER.debug("Round-Robin through loaders, Selected loader is {} ".format(loaders))
+        else:
+            loaders = self.loader_set.nodes
+        self.loaders = loaders
+
+        self.max_workers = len(loaders) * self.stress_num
+        LOGGER.debug("Starting %d %s Worker threads", self.max_workers, self.__class__.__name__)
+        self.executor = concurrent.futures.ThreadPoolExecutor(  # pylint: disable=consider-using-with
+            max_workers=self.max_workers)
+
+    def run(self):
+        self.configure_executer()
+        for loader_idx, loader in enumerate(self.loaders):
+            for cpu_idx in range(self.stress_num):
+                self.results_futures += [self.executor.submit(self._run_stress, *(loader, loader_idx, cpu_idx))]
+
+        return self
+
+    def _run_stress(self, loader, loader_idx, cpu_idx):
+        raise NotImplementedError()
+
+    def get_results(self):
+        results = []
+        timeout = self.timeout + 120
+        LOGGER.debug('Wait for %s stress threads results', self.max_workers)
+        for future in concurrent.futures.as_completed(self.results_futures, timeout=timeout):
+            results.append(future.result())
+
+        return results
+
+    def verify_results(self):
+        results = []
+        errors = []
+        timeout = self.timeout + 120
+        LOGGER.debug('Wait for %s stress threads to verify', self.max_workers)
+        for future in concurrent.futures.as_completed(self.results_futures, timeout=timeout):
+            results.append(future.result())
+
+        return results, errors
```

Comment:
will errors always empty?

Reply:
this code is just move from one file to other,
but it's was basic just to align the expected API out of this function, we don't parse the whole log at the end nowadays, we raise exception while running out of the logs of stress commands.